addons/web/static/src/views/form/form_controller.scss

```diff
@@ -471,6 +471,20 @@
                 width: auto;
             }
         }
+
+        .o_field_widget {
+            &.o_field_many2one_avatar {
+                display: block;
+            }
+
+            &.o_field_many2many_tags_avatar {
+                display: inline;
```

Comment:
There is a CSS custom property for that: `--fieldWidget-display`

Reply:
`--fieldWidget-display` CSS custom property is setting the display property as `inline-block` which is not suitable for removing flicker.